anvill/src/Lifters/FunctionLifter.cpp

```diff
@@ -260,16 +290,106 @@ void FunctionLifter::VisitDirectJump(const remill::Instruction &inst,
   llvm::BranchInst::Create(GetOrCreateTargetBlock(inst.branch_taken_pc), block);
 }
 
-// Visit an indirect jump control-flow instruction. This may be register- or
-// memory-indirect, e.g. `jmp rax` or `jmp [rax]` on x86. Thus, the target is
-// not know a priori and our default mechanism for handling this is to perform
-// a tail-call to the `__remill_jump` function, whose role is to be a stand-in
-// something that enacts the effect of "transfer to target."
+// This is a list of the possibilities we want to cover:
+//
+// 1. No target: AddTerminatingTailCall
+// 2. Single target, complete: normal jump
+// 3. Multiple targets, complete: switch with no default case
+// 4. Single or multiple targets, not complete: switch with default case
+//    containing AddTerminatingTailCall
+//
+// See the function declaration in the header file for more information
 void FunctionLifter::VisitIndirectJump(const remill::Instruction &inst,
                                        remill::Instruction *delayed_inst,
                                        llvm::BasicBlock *block) {
   VisitDelayedInstruction(inst, delayed_inst, block, true);
-  remill::AddTerminatingTailCall(block, intrinsics.jump);
+
+  // Attempt to get the target list for this control flow instruction
+  // so that we can handle this jump in a less generic way
+  auto maybe_target_list =
+      options.ctrl_flow_provider->TryGetControlFlowTargets(inst.pc);
+
+  auto add_remill_jump{true};
+  llvm::BasicBlock *current_bb = block;
+
+  if (maybe_target_list.has_value()) {
+    const auto &target_list = maybe_target_list.value();
+
+    if (target_list.destination_list.size() == 1U && target_list.complete) {
+
+      // If the target list is complete and has only one destination, then we
```

Comment:
I prefer putting these types of comments before the `if` or the `else if`, i.e. the comment describes the condition and the body if the condition is true. The code formatting script has stuff to properly handle this commenting style.